sql/catalyst/src/main/java/org/apache/spark/sql/connector/expressions/GeneralScalarExpression.java

```diff
@@ -190,6 +113,30 @@ public GeneralScalarExpression(String name, Expression[] children) {
   public String name() { return name; }
   public Expression[] children() { return children; }
 
+  @Override
+  public NamedReference[] references() {
+    return Arrays.stream(children()).map(e -> {
+      if (e instanceof NamedReference) {
```

Comment:
why do we need this check? Following the catalyst `Expression`, we just need to combine the references from children and deduplicate them.

Reply:
`Arrays.stream(children()).map(e -> e.references())`